### PR TITLE
Upgrade az vmware extension to use 2023-03-01 API version

### PR DESCRIPTION
**Pull request by kairu-ms**
---

* `az vmware`: update to AVS 2023-03-01 API
* `az vmware private-cloud create`: Add new `--ext-nw-blocks` parameter
*  `az vmware private-cloud update`: Add new `--ext-nw-blocks` parameter

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by kairu-ms** [comment]
Hi @jonathanhe-msft and @cataggar. I've changed the related the code. Could you help update the test recordings in your mock servers. Thanks.

File: src/vmware/azext_vmware/tests/latest/test_vmware_scenario.py [hunk 1]
```diff
@@ -140,6 +140,18 @@ def test_vmware_private_cloud(self):
...
+
+        # update the second extended network blocks
+        self.cmd('vmware private-cloud update -g {rg} -n {privatecloud} --ext-nw-blocks [1]=10.2.8.0/23')
+
+        # delete the second extended network blocks
+        self.cmd('vmware private-cloud update -g {rg} -n {privatecloud} --ext-nw-blocks [1]=null')
+
+        # delete all extended network blocks
+        self.cmd('vmware private-cloud update -g {rg} -n {privatecloud} --ext-nw-blocks null')
+
```
Comment (kairu-ms) [inline src/vmware/azext_vmware/tests/latest/test_vmware_scenario.py:154]: This is the recommended way to add, update and remove sub property by using `update` command.

**Comment by jonathanhe-msft** [comment]
`az vmware placement-policy vm-host` and `az vmware placement-policy vm` aren't working properly as they're missing a required affinityType parameter: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/vmware/resource-manager/Microsoft.AVS/stable/2023-03-01/vmware.json#L7718

**Comment by jonathanhe-msft** [comment]
`az vmware hcx-enterprise-site-create` failing due to bad request. the request looks fine to me, but the mock server isn't accepting the values. I think it's an error with the mock, but can you confirm everything looks ok from your end?

image

**Comment by jonathanhe-msft** [comment]
`vmware private-cloud add-identity-source` fails with `ResourceNotFoundError`, even though i see the calls made successfully to our mock server:

image
image
image

**Comment by kairu-ms** [comment]
> aren't working properly as they
[jonathanhe-msft](https://github.com/jonathanhe-msft) Could you use the `--debug` argument to check the api response. It seems the `affinityType` isn't returned in the api response of Get method. The update command will GET the resource first. And as `affinityType` is not exposed as an argument, the value will be inherent from the GET response. According to api specs, that property should be included in api response.

**Comment by kairu-ms** [comment]
> `az vmware hcx-enterprise-site-create` failing due to bad request. the request looks fine to me, but the mock server isn't accepting the values. I verified with the old implementation (pre-migration) and it is working as expected. can you take a look?
> 
> image

Fixed in this commit(https://github.com/Azure/azure-cli-extensions/pull/6765/commits/7d7faac452443a57c953f1806df6300074db8cd3)

**Comment by kairu-ms** [comment]
> `vmware private-cloud add-identity-source` fails with `ResourceNotFoundError`, even though i see the calls made successfully to our mock server:
> 
> image image image

That's because the 

> `vmware private-cloud add-identity-source` fails with `ResourceNotFoundError`, even though i see the calls made successfully to our mock server:
> 
> image image image

That's because the newly added identity source is missed in the response of your mock server. I also find this issue in old recordings and I manually fixed the server response in previous migration PR.

**Comment by jonathanhe-msft** [comment]
> > `az vmware placement-policy vm-host` and `az vmware placement-policy vm` aren't working properly as they're missing a required affinityType parameter: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/vmware/resource-manager/Microsoft.AVS/stable/2023-03-01/vmware.json#L7718
> > image
> 
> [jonathanhe-msft](https://github.com/jonathanhe-msft) Could you use the `--debug` argument to check the api response. It seems the `affinityType` isn't returned in the api response of Get method. The update command will GET the resource first. And as `affinityType` is not exposed as an argument, the value will be inherent from the GET response. According to api specs, that property should be included in api response.

confirmed get response is missing the `affinityType` property. working on fixing and will update live tests once complete.

**Comment by kairu-ms** [comment]
@zhoxing-ms Please help review this PR.

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**